include/range/v3/view/iota.hpp

```diff
@@ -163,6 +163,25 @@ namespace ranges
                     :  static_cast<D>(v0 - v1);
             }
 
+            template<typename Val, CONCEPT_REQUIRES_(SignedIntegral<Val>())>
+            RANGES_CXX14_CONSTEXPR
+            iota_difference_t<Val> ints_distance_(Val from, Val to) noexcept {
+                using D = iota_difference_t<Val>;
+                RANGES_EXPECT(from <= to);
+                RANGES_EXPECT((static_cast<D>(to) - static_cast<D>(from))
+                              < std::numeric_limits<D>::max());
```

Comment:
What does this expectation accomplish? If `to - from == max`, it's representable. `to - from <= max` always holds.